### PR TITLE
[Full Team Sign Off] Rubocop adjustments

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

Per discussion in dev planning 2020-11-17, here is a PR to increase the class length max and the method length max for rubocop.  
- the first commit is a refactor to slightly reorganize the rubocop.yml file (for ease of maintenance);  
- the second commit is the rule changes
- the third commit is the inline comments that must be removed with these rule changes
- the fourth commit is regenerating rubocop_todo (by hand, so maxes aren't increased)

Weigh in below, check the box when you've reviewed the PR at least once.  If there's lively discussion below, we'll move this to an infradev meeting
- [x] @aaron-collier 
- [x] @jermnelson 
- [x] @jmartin-sul 
- [x] @justinlittman 
- [x] @jcoyne 
- [x] @mjgiarlo 
- [x] @peetucket

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -6,45 +6,61 @@ require:
...
     - db/**/*
     - node_modules/**/*
     - bin/**/*
     - vendor/**/*
 
+Lint/MissingSuper: # (new in 0.89)
+  Enabled: true
+  Exclude:
+    - app/components/**/*
+
```
Comment (ndushay) [inline .rubocop.yml:22]: moved up to alphabetize

File: .rubocop.yml [hunk 2]
```diff
@@ -6,45 +6,61 @@ require:
...
+  Max: 150
+
+Metrics/MethodLength:
+  Max: 20
+
 RSpec/DescribeClass:
   Exclude:
     - spec/features/**/*
     - spec/requests/**/*
+
```
Comment (ndushay) [inline .rubocop.yml:37]: spaced them out for easier skimming

File: .rubocop.yml [hunk 3]
```diff
@@ -6,45 +6,61 @@ require:
...
-    - app/components/**/*
+#### Turning on New Rules
 
 Layout/BeginEndAlignment: # (new in 0.91)
   Enabled: true
 Layout/EmptyLinesAroundAttributeAccessor: # (new in 0.83)
   Enabled: true
 Layout/SpaceAroundMethodCallOperator: # (new in 0.82)
   Enabled: true
+
```
Comment (ndushay) [inline .rubocop.yml:63]: attempt to clearly delineate between different types of cops

File: .rubocop.yml [hunk 4]
```diff
@@ -53,10 +69,14 @@ Lint/DeprecatedOpenSSLConstant: # (new in 0.84)
   Enabled: true
 Lint/DuplicateElsifCondition: # (new in 0.88)
   Enabled: true
+Lint/DuplicateRegexpCharacterClassElement: # (new in 1.1)
+  Enabled: true
```
Comment (ndushay) [inline .rubocop.yml:73]: alphabetizing ...

File: .rubocop.yml [hunk 5]
```diff
@@ -6,45 +6,61 @@ require:
...
+
 Metrics/BlockLength:
   Exclude:
     - spec/**/*
 
+Metrics/ClassLength:
+  Max: 150
+
+Metrics/MethodLength:
+  Max: 20
```
Comment (ndushay) [inline .rubocop.yml:31]: cops actually changed in this PR

**Review by peetucket — APPROVED** [review]

**Review by jcoyne — APPROVED** [review]

**Review by jermnelson — APPROVED** [review]

File: .rubocop.yml (hunk 5 above)
Comment (aaron-collier) [reply .rubocop.yml:31]: For me, 20 pushes the limits of readability. I'm ok with it here, as we've discussed, but I'd prefer we avoid making it our "standard".

**Review by aaron-collier — APPROVED** [review]

File: .rubocop.yml (hunk 5 above)
Comment (ndushay) [reply .rubocop.yml:31]: @aaron-collier make me an offer.

Comment (aaron-collier) [reply .rubocop.yml:31]: I'd be happy at 15.

Comment (ndushay) [reply .rubocop.yml]: Just added a commit with method length max at 15.  It restores two inline exceptions for method length.  I can live with it ... unless we get a lot more of those for methods that have no clear refactor.

Comment (mjgiarlo) [reply .rubocop.yml]: I wouldn't die on this hill but... I'll ask: how would folks feel about 10?

File: .rubocop_todo.yml [hunk 6]
```diff
@@ -6,35 +6,20 @@
 # Note that changes in the inspected code, or installation of new
 # versions of RuboCop, may require this file to be generated again.
 
-# Offense count: 3
+# Offense count: 2
```
Comment (mjgiarlo) [inline .rubocop_todo.yml:9]: Are the hand edits here the same as we'd get from `rubocop --auto-gen-config --auto-gen-only-exclude` ?

**Review by mjgiarlo — COMMENTED** [review]
Overall, looking good! Have a couple questions.

File: .rubocop_todo.yml (hunk 6 above)
Comment (ndushay) [reply .rubocop_todo.yml:9]: nope - it changes the max in some with that option.  that's why I ended up doing them by hand.

File: .rubocop.yml (hunk 5 above)
Comment (ndushay) [reply .rubocop.yml]: 10 is too small for me.  it is the default;  this is me wanting to increase the default.

Comment (ndushay) [reply .rubocop.yml]: or Mike, are you saying you would prefer to keep it at 10, but you could compromise?  While I am saying I would prefer it to be 20, and I have compromised down to 15 ... but we are not agreeing yet?

Comment (mjgiarlo) [reply .rubocop.yml]: @ndushay I thought the default was 5, not 10. :laughing: let's stick with 15 for now.

**Review by mjgiarlo — APPROVED** [review]

**Review by justinlittman — APPROVED** [review]

**Review by jmartin-sul — APPROVED** [review]

**Pull request merged**